pkg/ingester/ingester_test.go

```diff
@@ -375,3 +375,42 @@ func benchmarkIngesterSeriesCreationLocking(b *testing.B, parallelism int) {
 
 	wg.Wait()
 }
+
+func BenchmarkIngesterPush(b *testing.B) {
+	cfg := defaultIngesterTestConfig()
+
+	const (
+		series  = 100
+		samples = 100
+	)
+
+	// Construct a set of realistic-looking samples, all with slightly different label sets
+	labels := chunk.BenchmarkMetric.Clone()
+	ts := make([]client.PreallocTimeseries, 0, series)
+	for j := 0; j < series; j++ {
+		labels["cpu"] = model.LabelValue(fmt.Sprintf("cpu%02d", j))
+		ts = append(ts, client.PreallocTimeseries{
+			TimeSeries: client.TimeSeries{
+				Labels: client.ToLabelPairs(labels),
+				Samples: []client.Sample{
+					{TimestampMs: 0, Value: float64(j)},
+				},
+			},
+		})
+	}
+	ctx := user.InjectOrgID(context.Background(), "1")
```

Comment:
Might want to stick a b.ResetTimer in here to clear out any allocation counters.

Reply:
yeah, it's de minimus and I couldn't be bothered carefully re-running the benchmark on master

Reply:
I added the line.